src/jit/valuenum.cpp

```diff
@@ -3163,6 +3163,51 @@ void ValueNumStore::GetConstantBoundInfo(ValueNum vn, ConstantBoundInfo* info)
     }
 }
 
+//------------------------------------------------------------------------
+// IsVNArrLenUnsignedBound: Checks if the specified vn represents an
+//    expression such as "(uint)i < (uint)a.len" that imply that the
+//    array index is valid (0 <= i && i < a.len).
+//
+// Arguments:
+//    vn - Value number to query
+//    info - Pointer to a ArrLenUnsignedBoundInfo object to return
+//           information about the expression. Not populated if the
+//           vn expression isn't suitable (e.g. i <= a.len)
+
+bool ValueNumStore::IsVNArrLenUnsignedBound(ValueNum vn, ArrLenUnsignedBoundInfo* info)
```

Comment:
It's a bit odd that `IsVNArrLenUnsignedBound` takes an out `info` parameter but its siblings have seprate `IsFoo` and `GetFooInfo` methods.  If I'm following correctly, doing it this way makes more sense for this case because then you have the `ArrLenUnsignedBoundInfo` available in `optCreateJTrueBoundAssertion` and can avoid making the useless complementary assertion.  Would be nice to make sure that's written down somewhere, in the commit message if not in a comment here.